Lab 3 Apartments

VsevolodMu opened the pull request:
Макет: https://ru.pinterest.com/pin/284219426483572294/
#60 Ежков + Барышев
<img width="1920" alt="Снимок экрана 2024-04-19 в 8 14 54 PM" src="https://github.com/reznikovkg/vue-buttons/assets/118456964/f6ed679f-9349-4e80-b71a-328350a31a75">
<img width="1920" alt="Снимок экрана 2024-04-19 в 8 15 07 PM" src="https://github.com/reznikovkg/vue-buttons/assets/118456964/61298203-a343-4089-97fa-db0943ecfdbc">
<img width="1920" alt="Снимок экрана 2024-04-19 в 8 15 18 PM" src="https://github.com/reznikovkg/vue-buttons/assets/118456964/dc00d559-ed08-4d10-911f-d86f7b2c19b4">
<img width="1920" alt="Снимок экрана 2024-04-19 в 8 15 30 PM" src="https://github.com/reznikovkg/vue-buttons/assets/118456964/344a356d-e516-4004-a4a5-78a0942b4084">
<img width="1920" alt="Снимок экрана 2024-04-19 в 8 15 37 PM" src="https://github.com/reznikovkg/vue-buttons/assets/118456964/cea61ba2-ad66-4d2d-b44f-2be2b4a4bcec">
<img width="1920" alt="Снимок экрана 2024-04-19 в 8 16 31 PM" src="https://github.com/reznikovkg/vue-buttons/assets/118456964/5b50ca3a-4b21-4d6c-b59a-63c9a61cbed6">
<img width="1920" alt="Снимок экрана 2024-04-19 в 8 16 45 PM" src="https://github.com/reznikovkg/vue-buttons/assets/118456964/b178f225-e7b9-47cc-b8f4-959c139295af">
<img width="1920" alt="Снимок экрана 2024-04-19 в 8 16 51 PM" src="https://github.com/reznikovkg/vue-buttons/assets/118456964/cff7bad6-7501-4a01-81d1-a75010ce0b89">

VsevolodMu committed:
Add component Header + router for him

VsevolodMu committed:
Clean Code

Pulsar700 committed:
create HomeFooter.vue

Pulsar700 committed:
first fixes on HomeFooter.vue

Pulsar700 committed:
first fixes on HomeFooter.vue

Pulsar700 committed:
second fixes on HomeFooter.vue

VsevolodMu committed:
Pip

VsevolodMu committed:
Merge remote-tracking branch 'origin/blogs' into blogs

# Conflicts:
#	vue/src/components/pages/blogEzhkovBaryshev/pages/HomePageEzhkovBaryshev.vue

VsevolodMu committed:
Hot fixes

VsevolodMu committed:
Seatch component ready

VsevolodMu committed:
Не доделан слайдер

VsevolodMu committed:
hot fix

Pulsar700 committed:
FooterHome.vue created

Pulsar700 committed:
text created

VsevolodMu committed:
Add slider and Card

VsevolodMu committed:
hot fixes

VsevolodMu committed:
hot fixes

Pulsar700 committed:
BodyMain.vue created

VsevolodMu committed:
Add cards

VsevolodMu committed:
Merge remote-tracking branch 'origin/blogs' into blogs

VsevolodMu committed:
Добавлена реактивность для карточек

VsevolodMu committed:
Нужно доделать поиск

VsevolodMu committed:
Поиск работает

Pulsar700 committed:
created

VsevolodMu committed:
Почти всё работает кроме ссылки на карточку

Pulsar700 committed:
created

Pulsar700 committed:
fix

VsevolodMu committed:
Всё работает

Pulsar700 committed:
fix

Pulsar700 committed:
fixes

Pulsar700 committed:
fixes

Pulsar700 committed:
AddApartmentsCard.vue fixes

VsevolodMu committed:
Правки

VsevolodMu committed:
Merge remote-tracking branch 'origin/blogs' into blogs

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/BodyPriceCard.vue:
```diff
@@ -0,0 +1,154 @@
+      type: String,
+      require: 'true'
+    },
+    price: {
```
Карточка должна получать объект данных, а не все поля по отдельности. Зачем раздувать код на ровном месте?

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/FooterHome.vue:
```diff
@@ -0,0 +1,196 @@
+  margin-left: 339px;
+}
+
+.footer-logo__text-down-third {
```
Где вложенность?

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/HeaderHome.vue:
```diff
@@ -0,0 +1,92 @@
+    <Search/>
+    <div class="buttonPagination">
+      <div
+          @click="() => move(index)"
```
Порядок

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/HeaderHome.vue:
```diff
@@ -0,0 +1,92 @@
+  height: 1025px;
+}
+
+.buttonPagination {
```
Написано не по стилю

VsevolodMu committed:
Убрал излишнее кол-во передаваемых параметров заменив на объект + теперь передаю объект фора, а не метода как раньше.

VsevolodMu committed:
Изменил порядок + написал по стилю

VsevolodMu committed:
Уже получше

Pulsar700 committed:
FooterHome.vue fixes

VsevolodMu commented:
Исправили

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/BodyMain.vue:
```diff
@@ -0,0 +1,273 @@
+        <p class="apartments__cards-text">From a room for a night to a loft for as long like, there's a RoamStay for
+          every
+          occasion.</p>
+        <router-link :to="{ name: routeNames.ADDAPARTMENTSCARD }">
```
RouterLink

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/BodyMain.vue:
```diff
@@ -0,0 +1,273 @@
+import {RouteNames} from "@/router/routes";
+
+export default {
+  components: {BodyPriceCard},
```
Нет названия компонента

reznikovkg replied:
Компоненты подключаем каждый с новой строки

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/FooterHome.vue:
```diff
@@ -0,0 +1,185 @@
+  font-size: 17px;
+}
+
+.logo__text-down-second {
```
Наследование и вложенность

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/HeaderNav.vue:
```diff
@@ -0,0 +1,160 @@
+</script>
+
+<style scoped lang="less">
+
```
Привести в порядок

reznikovkg commented on vue/src/store/blogEzhkovBaryshev/BackgroundCardStore.js:
```diff
@@ -0,0 +1,31 @@
+
+const state = {
+  BackgroundCardStore: [
```
Странный нейминг

Pulsar700 committed:
some fixes

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/BodyMain.vue:
```diff
@@ -0,0 +1,276 @@
+
+        <div class="search-block">
+          <form class="search-block__form">
+            <input type="text" name="text" class="search-block__text" placeholder="Поиск" v-model="searchRequest">
```
Приоритет v-model

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/BodyMain.vue:
```diff
@@ -0,0 +1,276 @@
+
+    &-paragraf {
+      text-align: center;
+
```
Много косяков по стилям. Выровнить и почистить

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/BodyMain.vue:
```diff
@@ -0,0 +1,276 @@
+    }
+
+    &-image {
+      text-align-all: center;
```
Что это за свойство?

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/FooterHome.vue:
```diff
@@ -0,0 +1,183 @@
+  }
+}
+</style>
+<script>
```
Что это?

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/FooterHome.vue:
```diff
@@ -0,0 +1,183 @@
+  }
+}
+
+.right-list__item:nth-child(n+1) {
```
Наследование и вложенность

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/HeaderHome.vue:
```diff
@@ -0,0 +1,95 @@
+          @click="() => move(index)"
+      >
+        <ButtonPagination
+            :is-active="index === currentIndex"
```
Ради одного свойства не нужно разносить на строки

reznikovkg replied:
НЕ ИСПРАВЛЕНО!

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/HeaderHome.vue:
```diff
@@ -0,0 +1,95 @@
+  height: 1025px;
+}
+
+.buttonPagination__component {
```
Только kebab-case!

Pulsar700 committed:
some design fixes

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/BodyMain.vue:
```diff
@@ -0,0 +1,272 @@
+        </div>
+        <div class="service-information__card">
+          <img class="service-information__card-image" src="../images/image-Photoroom.png-Photoroom-2.png"
+               alt="Логотип сайта">
```
Делать переносы правильно!

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/FooterHome.vue:
```diff
@@ -0,0 +1,180 @@
+</template>
+<style>
+
+.footer-wrapper {
```
Переписать стили футера с наследованием и вложенностью

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/FooterHome.vue:
```diff
@@ -0,0 +1,180 @@
+}
+
+.number {
+  margin-top: 1px;
```
Одинаковые стили, заменить на один общий. Такие отступы margin-left говорят о неверном использовании

Pulsar700 committed:
1 fix

VsevolodMu committed:
Внесены правки

VsevolodMu committed:
Merge remote-tracking branch 'origin/blogs' into blogs

# Conflicts:
#	vue/src/components/pages/blogEzhkovBaryshev/components/BodyMain.vue
#	vue/src/components/pages/blogEzhkovBaryshev/components/FooterHome.vue

VsevolodMu committed:
Внесены правки

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/pages/HomePageEzhkovBaryshev.vue:
```diff
@@ -0,0 +1,24 @@
+import FooterHome from "@/components/pages/blogEzhkovBaryshev/components/FooterHome.vue";
+import BodyMain from "@/components/pages/blogEzhkovBaryshev/components/BodyMain.vue";
+
+export default {
```
Названия компонентов

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/pages/ApartmentsPage.vue:
```diff
@@ -0,0 +1,94 @@
+    <HeaderNav/>
+    <div class="apartment__card" :style="cardImage">
+      <h2 class="apartment__card-name">
+        {{ getApartmentsCardStore[$route.params.id]['name'] }}
```
Вынести в `computed` `getApartmentsCardStore[$route.params.id]`

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/pages/AddApartmentsCard.vue:
```diff
@@ -0,0 +1,149 @@
+    ]),
+    create() {
+      if (this.id === '' || this.name === '' || this.price === '' || this.guests === '' || this.bedrooms === '' || this.bathrooms === '' || this.place === '' || this.image === '')
+        alert('Данные введены некоректно')
```
Блок {}

reznikovkg replied:
this.form = {}
Object.values(this.form).some(i => !i)

reznikovkg commented on vue/src/components/pages/blogEzhkovBaryshev/components/FooterHome.vue:
```diff
@@ -0,0 +1,194 @@
+          width: 40px;
+        }
+
+        & li {
```
Не стилизовать теги

VsevolodMu committed:
Вынес в computed

VsevolodMu committed:
Избавился от стилизации дефолтных тегов.
Но оставил сброс стилий глобально для этого файла по той причине, что прошлая верстка уже основана на дефолтных значениях тегов, поэтому если вынести в глобал, то поедет верстка.
В других местах исправлено.
+ Добавил название компонентаю.

reznikovkg commented:
Сдано

VsevolodMu committed:
Add express + cors

VsevolodMu committed:
Add axios + storage for index.js

VsevolodMu committed:
Пагинация работает

VsevolodMu committed:
Все работает
